pkg/apis/app/v1beta1/application_types.go

```diff
@@ -84,15 +84,36 @@ type ApplicationStatus struct {
 
 // ImageSpec contains information about an image used as an icon.
 type ImageSpec struct {
-	// The source for image represented as either an absolute URL to the image or a Data URL containing
+	// Binary contains the base64-encoded image binary.
+	//
+	// Only one of Binary, Text, URL, or ConfigMapKeyRef may be defined.
+	Binary string `json:"binary,omitempty"`
+
+	// Text contains the image definition as plain text, for formats such as SVG.
+	//
+	// Only one of Binary, Text, URL, or ConfigMapKeyRef may be defined.
+	Text string `json:"text,omitempty"`
+
+	// URL for the image, either as an absolute URL to the image or a Data URL containing
 	// the image. Data URLs are defined in RFC 2397.
-	Source string `json:"src"`
+	//
+	// Only one of Binary, Text, URL, or ConfigMapKeyRef may be defined.
+	URL string `json:"url,omitempty"`
 
-	// (optional) The size of the image in pixels (e.g., 25x25).
+	// Size of the image in pixels (e.g., 25x25). Optional.
 	Size string `json:"size,omitempty"`
 
-	// (optional) The mine type of the image (e.g., "image/png").
+	// Type indicates the MIME type of the image (e.g., "image/png"). Optional.
 	Type string `json:"type,omitempty"`
+
+	// ConfigMapKeyRef is a reference to a ConfigMap entry containing the image data.
+	//
+	// Only one of Binary, Text, URL, or ConfigMapKeyRef may be defined.
+	ConfigMapKeyRef *ConfigMapKeySelector `json:"configMapKeyRef,omitempty"`
```

Comment:
If someone were going to build a UI to display images in a configmap, how would they do it? A URL or Data URL are something that can be passed to a browser or other viewer. How would someone use a file in a configmap?

Reply:
They would likely need to manually convert the info to a data URL - this is what the OpenShift (formerly Tectonic) console does under the covers. See [here](https://github.com/openshift/console/blob/master/frontend/public/components/cloud-services/index.tsx#L190).

Reply:
Can we be more specific about the format of the ConfigMap value? Eg. "[...] containing JSON-formatted ImageSpec to use"

Reply:
I don't see the advantages to do this via configmap, the definition of an application is not tied to be shipped with a configmap holding the image. The image could already be inserted as base64 data for offline usage.